data/routes/lib/json.lua

```diff
@@ -0,0 +1,389 @@
+-- https://github.com/rxi/json.lua
+--
+-- json.lua
+--
+-- Copyright (c) 2020 rxi
+--
+-- Permission is hereby granted, free of charge, to any person obtaining a copy of
+-- this software and associated documentation files (the "Software"), to deal in
+-- the Software without restriction, including without limitation the rights to
+-- use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies
+-- of the Software, and to permit persons to whom the Software is furnished to do
+-- so, subject to the following conditions:
+--
+-- The above copyright notice and this permission notice shall be included in all
+-- copies or substantial portions of the Software.
+--
+-- THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+-- IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+-- FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+-- AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+-- LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+-- OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+-- SOFTWARE.
+--
+
+local json = { _version = "0.1.2" }
+
+-------------------------------------------------------------------------------
+-- Encode
+-------------------------------------------------------------------------------
+
+local encode
+
+local escape_char_map = {
+	[ "\\" ] = "\\",
+	[ "\"" ] = "\"",
+	[ "\b" ] = "b",
+	[ "\f" ] = "f",
+	[ "\n" ] = "n",
+	[ "\r" ] = "r",
+	[ "\t" ] = "t",
```

Comment:
This whole file is a library straight from https://github.com/rxi/json.lua
I think it would be unwise to reformat it, as we wouldn't be able to diff it in future (and git submodule isn't worth the tiny hassle for just one file either), but I think it makes sense to exempt it from our formatting rules.

Reply:
why is this map needed? Can't we just use the same function we use to escape strings in sql?

Reply:
I don't know, this is an external library that is copied into, it's not meant to be reviewed or altered in any way. We can choose a different one if there's a better one.